tracer/src/Datadog.Trace/AppSec/Rasp/RaspModule.cs

```diff
@@ -0,0 +1,55 @@
+// <copyright file="RaspModule.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System.Collections.Generic;
+using Datadog.Trace.AppSec.Coordinator;
+using Datadog.Trace.AppSec.Waf;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Telemetry;
+using static Datadog.Trace.Telemetry.Metrics.MetricTags;
+
+namespace Datadog.Trace.AppSec.Rasp;
+
+internal static class RaspModule
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(RaspModule));
+
+    internal static void OnLfi(string file)
+    {
+        CheckVulnerability(AddressesConstants.FileAccess, file);
+    }
+
+    private static void CheckVulnerability(string address, string valueToCheck)
+    {
+        var security = Security.Instance;
+
+        if (!security.RaspEnabled)
+        {
+            return;
+        }
+
+        var rootSpan = Tracer.Instance.InternalActiveScope?.Root?.Span;
+
+        if (rootSpan is null)
+        {
+            return;
+        }
+
+        var arguments = new Dictionary<string, object> { [address] = valueToCheck };
+        RunWaf(arguments, rootSpan, address, valueToCheck);
+    }
+
+    private static IResult? RunWaf(Dictionary<string, object> arguments, Span rootSpan, string address, string value)
+    {
+        var securityCoordinator = new SecurityCoordinator(Security.Instance, SecurityCoordinator.Context, rootSpan);
```

Comment:
as `HttpContext` is now a property of the `SecurityCoordinator`, I think we can remove it from the ctor of `SecurityCoordinator`